### ISSUE TITLE
parallel_offset adding an additional point on the opposite side under certain circumstances

### ISSUE THREAD

**Issue by bentheiii** [shapely/shapely#690]
## Expected behavior and actual behavior.

under some very specific situations, `parallel_offset` inserts an additional point on the opposite side of the original line at the start or end (and sometimes both).

So far I have observed this occurring if the first and last points have equal `y` coordinates, and the first and last lines are parallel.

```python
base = LineString([(0, 1.2), (0, 1.5), (1, 1), (1, 1.2)])  # in black
offset = base.parallel_offset(0.1, 'left', join_style=1)  # in blue
```

![image](https://user-images.githubusercontent.com/13264001/54926021-27f9c300-4f18-11e9-9ce2-8a6a6531c70a.png)

```python
offset = base.parallel_offset(0.1, 'right', join_style=1)  # in blue
```
![image](https://user-images.githubusercontent.com/13264001/54926132-5e374280-4f18-11e9-9f1d-bff6f169e3a2.png)

## Operating system

windows 10 x64

## Shapely version and provenance

1.6.4.post1 installed from PyPI using pip

**Comment by sgillies** [comment]
@bentheiii this is not good, but not something we can directly fix in shapely. The algorithm is contained in GEOS, a C/C++ library, and will need to be fixed there: https://github.com/libgeos/geos. You could do the shapely project a big favor by submitting a report upstream there. If you can't, I'll make a note to do so at the end of the week.

**Comment by sgillies** [comment]
I'm going to close this. If it wasn't fixed in GEOS 3.9, let's reopen.

**Comment by jorisvandenbossche** [comment]
Confirming this works correctly now (with latest released GEOS):

```
base = LineString([(0, 1.2), (0, 1.5), (1, 1), (1, 1.2)])  # in black
offset = base.parallel_offset(0.1, 'left', join_style=1)  # in blue

from shapely.plotting import plot_line
plot_line(base, color="k")
plot_line(offset, color="b")
```

![shapely_offset1](https://user-images.githubusercontent.com/1020496/205436128-fb70955c-9050-4a14-9c00-5ac152ae4ee4.png)

and second case:

```
offset = base.parallel_offset(0.1, 'right', join_style=1)  # in blue
```

![shapely_offset2](https://user-images.githubusercontent.com/1020496/205436155-01063f71-e18d-43f7-a67d-5eb13515c1d3.png)
